File: 495. Teemo Attacking.java
public class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int cnt = duration, l = timeSeries.length - 1;
        if (l == -1) {
            return 0;
        }
        for (int i = 0; i < l; i++) {
            if (timeSeries[i] + duration < timeSeries[i + 1]) {
                cnt += duration;
            } else {
                cnt += (timeSeries[i + 1] - timeSeries[i]);
            }
        }
        return cnt;
    }
}

File: 495. Teemo Attacking.go
func findPoisonedDuration(timeSeries []int, duration int) int {
    if len(timeSeries) == 0 {
        return 0
    }
    var cnt = duration
    for i := 0; i < len(timeSeries) - 1; i++ {
        if timeSeries[i] + duration < timeSeries[i + 1] {
            cnt += duration
        } else {
            cnt += (timeSeries[i + 1] - timeSeries[i])
        }
    }
    return cnt
}
